### DISCUSSION TITLE
Policy for facet distribution

### DISCUSSION THREAD

**Discussion by gmourier** [meilisearch/product#315] [Feedback & Feature Proposal]
With `v0.21.0` version, we have removed facet distribution with 0 values matching the filters.

Before the `v0.21` version, facets matching 0 documents after a filter were returned. We could propose an option to choose the desired behavior for the facet distribution.

How would you interact with the API? What could the field be called at search? Could it be interesting to have it as an instance setting?

**Comment by paligiannis** [comment]
Hi @gmourier ! As @bidoubiwa mentioned in here https://github.com/meilisearch/instant-meilisearch/pull/696 it seems that having the items returning 0 even a facet is not selected is not that useful. 
Firstly, keeping facets disjunctive is really important for facets based on different item requirements. For example in e-commerce platforms you can select a facet with the name "gbOfRam". User should be able to see all items fitting his needs.

**Comment by ar-siddiqui** [comment]
In our use case, we do not like the meilisearch API not returning 0 facets. We want to show users to see 0 value facets as well so that they can select them. The hack provided by this [PR](https://github.com/meilisearch/instant-meilisearch/pull/695) only works when you first do a no filter search, if you directly use a URL with specific rules it would not showcase them as the client does not have anything to remember from.

IMO the meilisearch API should always return 0 facets (or be configurable to return 0 facets) and the client could be configured to either display them or not.

For example, displaying 0 facets allow a user to add `spreadsheet` as well to the search
![image](https://user-images.githubusercontent.com/53625184/159548177-fb5892da-2730-4548-8c61-55394af5d28c.png)

But without 0 facets, the user is not even aware that a type `spreadsheet` exists.
![image](https://user-images.githubusercontent.com/53625184/159548495-135711aa-ddd7-43d3-b9f1-9fc7f7c1f358.png)

**Reply by paligiannis** [reply]
exactly, what i was thinking of!

**Reply by paligiannis** [reply]
@bidoubiwa is a lifesaver
